packages/babel-generator/src/node/parentheses.js

```diff
@@ -55,6 +55,13 @@ export function ObjectExpression(node: Object, parent: Object): boolean {
     return true;
   }
 
+  if ((t.isBinaryExpression(parent) || t.isLogicalExpression(parent)) && parent.left === node) {
+    // We'd need to check that the parent's parent is an ExpressionStatement. But this
```

Comment:
More than just an `ExpressionStatement`, it could be an `AssignmentExpression` as well.

```js
foo = {} === 1;
```